### ISSUE TITLE
fmt::formatter for std::exception causes compile error when used with incomplete types

### ISSUE THREAD

**Issue by piotrrak** [fmtlib/fmt#3573]
Please take a look at hard compile error caused by `formatter` for `std::exception`:
https://godbolt.org/z/7chTPYed8

The bit more tricky to fix way would be
shield from incomple types `enable_if<conjunction<is_complete<T>, is_base_of<...>>>`
   but is_complete traits is tricky to get right between compiler versions
Or
 simply provide `formatter<std::exception>`; I can't see the reason why use SFINAE her, am I missing something?

Formatting incomplete types is really really useful;
Only the formatter must have complete definition some way.

**Comment by vitaut** [comment]
I don't think we can do much about it. Only providing a formatter for `exception` won't handle subclasses and defining formatters for every subclass is impractical. You could probably use `is_complete` in your SFINAE/concept check but it will likely be an ODR footgun.

**Comment by piotrrak** [comment]
Right, maybe it could be fixed at least for c++20 -  `__cpp_concepts >= 201907L` should be suffiecient to constrain without UB. `requires requires { sizeof(T) => 0; }` will SFINAE reject such case; Please note there is no type instantiation here, thus no ODR violation.
